stanify/xarray_test.py

```diff
@@ -0,0 +1,140 @@
+# )
+# print(ds)
+# print(ds.dims['x'])
+#
+# from cmdstanpy import CmdStanModel, write_stan_json
+#
+# import arviz as az
+# import pandas as pd
+# import numpy as np
+# import xarray as xr
+# data = pd.read_csv("../data/linear_regression_data.csv", index_col=0)
+# time_since_joined = data.time.values
+# slack_comments = data.comments.values
+# github_commits = data.commits.values
+# names = data.index.values
+# N = len(names)
+#
+# print(slack_comments, github_commits, names)
+# prior_model = CmdStanModel(stan_file="../stan_files/arviz/linreg_prior.stan")
+# linreg_prior_data_dict = {"N": N, "time_since_joined": time_since_joined}
+# write_stan_json("linreg_prior.data.json", linreg_prior_data_dict)
+# prior_fit = prior_model.sample(data="linreg_prior.data.json", chains=2, iter_sampling=100)
+# idata_kwarg = dict(
+#     prior_predictive = ["slack_comments_hat", "github_commits_hat"],
+#     posterior_predictive = ["slack_comments_hat", "github_commits_hat"],
+#     coords = {"developer": names},
+#     dims = {
+#         "slack_comments":["developer"],
+#         "github_comments":["developer"],
+#         "slack_comments_hat": ["developer"],
+#         "github_comments_hat": ["developer"],
+#         "time_since_joined": ["developer"],
+#     }
+#
+# )
+# idata_kwargs = dict(
+#     prior_predictive=["slack_comments_hat","github_commits_hat"],
+#     posterior_predictive=["slack_comments_hat","github_commits_hat"],
+#
+#     coords={"developer": names},
+#     dims={
+#         "slack_comments": ["developer"],
+#         "github_commits" : ["developer"],
+#         "slack_comments_hat": ["developer"],
+#         "github_commits_hat": ["developer"],
+#         "time_since_joined": ["developer"],
+#     }
+# )
+# fun = az.from_cmdstanpy(prior= prior_fit, **idata_kwarg)
+# print(fun)
+
+# sbc = xr.open_dataset("../data/pp_S3R2/generator.nc")
+#
+# print(sbc)
+# print(sbc.posterior)
+# print(sbc.groups)
+
+precision ={
+    "S": 3, # # of draws from prior
+    "M": 100, # # of draws from posterior (# of chains * # of draws from each chain)
+    "N": 2, # # of observation
+    "Q": 3, # # of target_simulated_stock
+    "R": 3, # # of subgroups for hierarchical Bayes
+    "time_saveper": .03,
+    "integration_times": np.arange(0, 200) * .03 + 0.01,  # initial_time < integration_times, .03 need not be time_saveper, but length should be N
+}
+
+idata_kwargs = dict(
+    prior_predictive=["prey_obs", "predator_obs"],
+    posterior_predictive=["prey_obs_post", "predator_obs_post"],
+    log_likelihood={
+        "loglik": "loglik"
+    },
+    coords={
+        "time": [n for n in range(precision['N'])],
+        "stock": [q for q in range(precision['Q'])],
+    },
+    dims={
+        'initial_outcome': ["stock"],
+        'integrated_result': ["time", "stock"],
+        'prey': ["time"],
+        'predator': ["time"],
+        'process_noise': ["time"],
+        "prey_obs": ["time"],
+        "predator_obs": ["time"],
+    }
+)
+setting = {
```

Comment:
I recommend that instead of passing settings as a complex nested dict/list, Make them into arguments for `data2draws/draws2data/data2draws2data`. This format is frustrating to debug and handle.

Reply:
Could you have first go on this?